x-pack/legacy/plugins/lens/public/indexpattern_plugin/loader.ts

```diff
@@ -278,10 +279,12 @@ function fromSavedObject(
     id,
     type,
     title: attributes.title,
-    fields: (JSON.parse(attributes.fields) as IndexPatternField[]).filter(
-      ({ type: fieldType, esTypes }) =>
-        fieldType !== 'string' || (esTypes && esTypes.includes('keyword'))
-    ),
+    fields: (JSON.parse(attributes.fields) as IndexPatternField[])
+      .filter(
+        ({ type: fieldType, esTypes }) =>
+          fieldType !== 'string' || (esTypes && esTypes.includes('keyword'))
+      )
+      .concat(documentField),
```

Comment:
This is the one part of the change that doesn't make sense to me. Why add it to the index pattern?

Reply:
Because it allows us to put this in one place, and treat all fields in a uniform way. Basically, it reduces the number of places we have to special case for this, and simplified a fair number of rabbit-trails I ended up going down.